lib/fs.js

```diff
@@ -1762,6 +1766,15 @@ function symlink(target, path, type_, callback_) {
   const type = (typeof type_ === 'string' ? type_ : null);
   const callback = makeCallback(arguments[arguments.length - 1]);
 
+  if (permission.isEnabled()) {
+    // The permission model's security guarantees fall apart in the presence of
+    // relative symbolic links. Thus, we have to prevent their creation.
+    if (!isAbsolute(toPathIfFileURL(target))) {
```

Comment:
This applies to all: If you move this line after getValidatedPath, you'd reduce the number of C++ calls, since getValidatedPath also calls toPathIfFileUrl function.

Reply:
This is tricky. I didn't find any elegant solution. It cannot be after `getValidatedPath()` because `isAbsolute()` will always be true after `getValidatedPath()` returns.